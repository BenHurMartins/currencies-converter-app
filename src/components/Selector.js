import '../styles.css';

const Selector = (props) => {
    return (
        <select value={props.value} onChange={props.onChange}
        >
            {
                props.options.map(item => {
                    return <option value={item}>{props.currencies[item]}</option>
                })
            }
        </select>
    )
}

export default Selector;