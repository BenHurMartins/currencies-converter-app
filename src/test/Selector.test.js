import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Selector from "../components/Selector";
afterEach(cleanup);

test('Renders Selector.js correctly', () => {
  const currencies = {
    'USD': 'United States Dolar',
    'BRL': 'Brazilian Real'
  }
  const options = [
    'USD', 'BRL'
  ]
  let value = 'USD'

  const changeValue = (text) => { value = text }
  const component = renderer.create(<Selector value={value} onChange={changeValue} options={options} currencies={currencies} />)

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

