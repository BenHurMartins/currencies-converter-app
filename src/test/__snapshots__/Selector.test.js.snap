// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Selector.js correctly 1`] = `
<select
  onChange={[Function]}
  value="USD"
>
  <option
    value="USD"
  >
    United States Dolar
  </option>
  <option
    value="BRL"
  >
    Brazilian Real
  </option>
</select>
`;
